embedded/store/immustore.go

```diff
@@ -2566,35 +2566,31 @@ func (s *ImmuStore) ReadValue(entry *TxEntry) ([]byte, error) {
 }
 
 func (s *ImmuStore) readValueAt(b []byte, off int64, hvalue [sha256.Size]byte) (n int, err error) {
-	mustReadFromVLog := true
-
 	if s.vLogCache != nil {
 		val, err := s.vLogCache.Get(off)
 		if err == nil {
 			// the requested value was found in the value cache
 			copy(b, val.([]byte))
-			mustReadFromVLog = false
+			return len(b), nil
 		} else {
 			if !errors.Is(err, cache.ErrKeyNotFound) {
 				return 0, err
 			}
 		}
 	}
 
-	if mustReadFromVLog {
-		vLogID, offset := decodeOffset(off)
+	vLogID, offset := decodeOffset(off)
 
-		if vLogID > 0 {
-			vLog := s.fetchVLog(vLogID)
-			defer s.releaseVLog(vLogID)
+	if vLogID > 0 {
```

Comment:
I think we should return error is `vLogID === 0`

Reply:
makes sense, what error should it be?

Reply:
I think we could return the existing error `ErrOffsetOutOfRange`

Reply:
Closing this in favor of issue created for handling this condition